src/msg/Messenger.h

```diff
@@ -610,12 +610,14 @@ class Messenger {
    *  one reference to it.
    */
   void ms_deliver_dispatch(Message *m) {
-    m->set_dispatch_stamp(ceph_clock_now());
     for (list<Dispatcher*>::iterator p = dispatchers.begin();
 	 p != dispatchers.end();
 	 ++p) {
-      if ((*p)->ms_dispatch(m))
+      utime_t dispatch_stamp = ceph_clock_now();
+      if ((*p)->ms_dispatch(m)) {
+        m->set_dispatch_stamp(dispatch_stamp);
```

Comment:
Hi, joscollin. 
My change at L587 means that the `dispatch_stamp` will be set to the time of **before** the message is dispatched by function `ms_fast_dispatch()`. 
So change at L587 and here have the same logicl: set `dispatch_stamp`  to the time of **before** the message is dispatched.